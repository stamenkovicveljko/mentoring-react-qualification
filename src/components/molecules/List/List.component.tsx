import React from "react";
import * as Styled from "./List.styles";

interface Props {
  items: Array<React.ReactNode>;
}

function List(props: Props) {
  const { items = [] } = props;
  return (
    <Styled.Ul>
      {items.map((item, ind) => (
        <li key={ind}>{item}</li>
      ))}
    </Styled.Ul>
  );
}

export default List;
