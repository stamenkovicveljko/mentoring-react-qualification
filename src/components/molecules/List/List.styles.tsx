import styled from "styled-components";
import { resetUl } from "../../../styles/resets";

export const Ul = styled.ul`
  ${resetUl};

  > li {
    padding: 10px;
    &:not(:first-child) {
      border-top: 1px solid lightgray;
    }
  }
`;
