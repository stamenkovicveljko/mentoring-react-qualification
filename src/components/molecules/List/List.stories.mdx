import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import List from "./List.component";

<Meta title="List" component={List} />

# List Component

List component with basic styles.
Provide `items` - which can be any `JSX` array and it works.

## Implementation

<Preview>
  <Story name="default">
    <List items={["list item 1", "list item 2", "list item 3"]} />
  </Story>
</Preview>

As mentioned in the begining it can render any JSX elements.
E.g.

```jsx
<List
  items={Array.from({ length: 5 }).map((_, i) => (
    <span>JSX list item {i + 1} </span>
  ))}
/>
```

## Props

<Props of={List} />
