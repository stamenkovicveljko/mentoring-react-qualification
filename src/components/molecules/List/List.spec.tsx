import React from "react";
import renderer from "react-test-renderer";
import List from "./List.component";

test("List renders as expected", () => {
  const component = renderer.create(
    <List items={["first item", "second item"]} />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
