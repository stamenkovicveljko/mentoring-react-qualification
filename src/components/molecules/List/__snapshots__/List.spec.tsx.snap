// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders as expected 1`] = `
<ul
  className="sc-AxjAm kTtWTX"
>
  <li>
    first item
  </li>
  <li>
    second item
  </li>
</ul>
`;
